plugins/auto_merge/merge_status.go

```diff
@@ -0,0 +1,101 @@
+package auto_merge
+
+import (
+	"log"
+	"regexp"
+
+	"github.com/xanzy/go-gitlab"
+)
+
+const authorTag = "<!-- author:lassie-bot-dog -->"
+
+func getMergeStatusLevelEmojis(mergeStatusLevel mergeStatusLevel) string {
+	if mergeStatusLevel == mergeStatusSuccess {
+		return ":green_heart:"
+	}
+
+	if mergeStatusLevel == mergeStatusFailed {
+		return ":poop:" // :collision: :exclamation:
+	}
+
+	// mergeStatusLevel == mergeStatusSkipped
+	return ":hourglass_flowing_sand:"
+}
+
+func decodeMergeStatus(comment string) *mergeStatus {
+	// TODO
+	hasConflicts := mergeStatusFailed
+	openDicussions := mergeStatusSkipped
+	passingPipeline := mergeStatusSkipped
+	enoughApprovals := mergeStatusSkipped
+
+	return &mergeStatus{
+		hasConflicts:    hasConflicts,
+		openDicussions:  openDicussions,
+		passingPipeline: passingPipeline,
+		enoughApprovals: enoughApprovals,
+	}
+}
+
+func encodeMergeStatus(status *mergeStatus) string {
+	comment := authorTag + "\n"
+	comment = comment + "### Your current merge request status is:\n\n"
+	comment = comment + "- " + getMergeStatusLevelEmojis(status.hasConflicts) + " Your changes do not have conflicts with the target branch\n"
+	comment = comment + "- " + getMergeStatusLevelEmojis(status.openDicussions) + " All discussions about your changes have been resolved\n"
+	comment = comment + "- " + getMergeStatusLevelEmojis(status.passingPipeline) + " A pipline successfully tested your code\n"
+	comment = comment + "- " + getMergeStatusLevelEmojis(status.enoughApprovals) + " Enough reviewers liked your code\n"
```

Comment:
I think these messages are kind of misleading. I would propose a simple description like `conflicts with the target branch` or `discussions resolved` letting the icon be the only status indicator.

Another possibility would be changing messages when the status changes:

- :poop: Your changes are conflicting with the target branch
- :green_heart: Your changes do not have conflicts with the target branch

Reply:
@lukashass Sure we should further improve them 🙃. I like your second approach more as I would like to keep it thematic with some Lassie specialties 😉

Reply:
@lukashass Could you have a look at the latest changes. I added some changing status messages.

Reply:
@anbraten Pretty cool stuff :+1: